### PR TITLE
Support different networks

### PR DESCRIPTION
**Pull request by jmoreira-valory**
Support different networks through configuration file `configs/mechs.json`, and also overriding configuration using environment variables starting with `MECHX_`:

```
MECHX_LEDGER_ADDRESS
MECHX_LEDGER_CHAIN_ID
MECHX_LEDGER_POA_CHAIN
MECHX_LEDGER_DEFAULT_GAS_PRICE_STRATEGY
MECHX_LEDGER_IS_GAS_ESTIMATION_ENABLED

MECHX_CHAIN_RPC
MECHX_WSS_ENDPOINT
MECHX_GAS_LIMIT
MECHX_CONTRACT_ABI_URL
MECHX_SUBGRAPH_URL
```

## Breaking changes
Environment variable renaming:

```
MANUAL_GAS_LIMIT --> MECHX_GAS_LIMIT
WEBSOCKET_ENDPOINT --> MECHX_WSS_ENDPOINT
```

### CODE REVIEW

File: mech_client/configs/mechs.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
+    "gnosis": {
```
Comment (0xArdi) [inline mech_client/configs/mechs.json:2]: are you adding new networks here?

File: mech_client/interact.py [hunk 2]
```diff
@@ -423,12 +519,21 @@ def interact(  # pylint: disable=too-many-arguments,too-many-locals
...
-    ledger_api = EthereumApi(**LEDGER_CONFIG)
+    ledger_api = EthereumApi(**asdict(ledger_config))
 
-    tool = verify_or_retrieve_tool(agent_id=agent_id, ledger_api=ledger_api, tool=tool)
-    abi = get_abi(contract_address=contract_address)
+    tool = verify_or_retrieve_tool(
+        agent_id=agent_id,
+        ledger_api=ledger_api,
+        tool=tool,
+        agent_registry_contract=mech_config.agent_registry_contract,
```
Comment (DavidMinarsch) [inline mech_client/interact.py:530]: where is this used?

File: mech_client/cli.py [hunk 3]
```diff
@@ -112,6 +118,7 @@ def interact(  # pylint: disable=too-many-arguments
             retries=retries,
             timeout=timeout,
             sleep=sleep,
+            chain_id=chain_id,
```
Comment (jmoreira-valory) [inline mech_client/cli.py:121]: Perhaps this name is confusing. It refers to the ID entry in the mechs.json configuration file (which can be an arbitrary value: polygon, polygon-config-2, etc... )

File: mech_client/configs/mechs.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+            "chain_id": 100,
+            "poa_chain": false,
+            "default_gas_price_strategy": "eip1559",
+            "is_gas_estimation_enabled": false
+        },
+        "manual_gas_limit": 100000,
+        "block_explorer_url": "https://gnosis.blockscout.com/api/v2/smart-contracts/{contract_address}",
+        "subgraph_url": "https://api.studio.thegraph.com/query/57238/mech/version/latest"
+    }
+}
```
Comment (DavidMinarsch) [inline mech_client/configs/mechs.json:32]: missing: celo, optimism, base and arbitrum

File: mech_client/configs/mechs.json (hunk 1 above)
Comment (jmoreira-valory) [reply mech_client/configs/mechs.json:2]: Yes, what is the issue ?

File: mech_client/interact.py (hunk 2 above)
Comment (jmoreira-valory) [reply mech_client/interact.py:530]: To fetch the available tools.

File: mech_client/cli.py (hunk 3 above)
Comment (jmoreira-valory) [reply mech_client/cli.py]: Changed to chain_config

File: mech_client/configs/mechs.json (hunk 4 above)
Comment (jmoreira-valory) [reply mech_client/configs/mechs.json:92]: Added placeholders for them. @0xArdi please fill when the data is available.

File: mech_client/configs/mechs.json [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        "wss_endpoint": "wss://polygon-bor-rpc.publicnode.com",
+        "ledger_config": {
+            "address": "https://polygon-bor-rpc.publicnode.com",
+            "chain_id": 137,
+            "poa_chain": false,
+            "default_gas_price_strategy": "eip1559",
+            "is_gas_estimation_enabled": false
+        },
+        "gas_limit": 100000,
+        "contract_abi_url": "",
```
Comment (jmoreira-valory) [inline mech_client/configs/mechs.json:44]: @0xArdi please fill.

File: mech_client/configs/mechs.json [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+        "wss_endpoint": "wss://optimism-rpc.publicnode.com",
+        "ledger_config": {
+            "address": "https://mainnet.optimism.io",
+            "chain_id": 10,
+            "poa_chain": false,
+            "default_gas_price_strategy": "eip1559",
+            "is_gas_estimation_enabled": false
+        },
+        "gas_limit": 100000,
+        "contract_abi_url": "",
```
Comment (jmoreira-valory) [inline mech_client/configs/mechs.json:89]: @0xArdi please fill.

**Review by jmoreira-valory — COMMENTED** [review]
NOTE: All subgraphs endpoint are wrong (except for gnosis). Therefore, it will not work until fixed.

File: mech_client/configs/mechs.json (hunk 6 above)
Comment (DavidMinarsch) [inline mech_client/configs/mechs.json:89]:
```suggestion
        "contract_abi_url": "https://api-optimistic.etherscan.io/api?module=contract&action=getabi&address={contract_address}",
```

File: mech_client/configs/mechs.json [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+        "ledger_config": {
+            "address": "https://arbitrum.llamarpc.com",
+            "chain_id": 42161,
+            "poa_chain": false,
+            "default_gas_price_strategy": "eip1559",
+            "is_gas_estimation_enabled": false
+        },
+        "gas_limit": 100000,
+        "contract_abi_url": "https://api.arbiscan.io/api?module=contract&action=getabi&address={contract_address}",
+        "subgraph_url": "https://api.studio.thegraph.com/query/57238/mech/version/latest"
```
[the remark this answers is not in the document]
Comment (DavidMinarsch) [reply mech_client/configs/mechs.json:30]: https://github.com/valory-xyz/mech-client/pull/32#pullrequestreview-1966705538

File: mech_client/configs/mechs.json [hunk 8]
```diff
@@ -27,7 +27,7 @@
         },
         "gas_limit": 100000,
         "contract_abi_url": "https://api.arbiscan.io/api?module=contract&action=getabi&address={contract_address}",
-        "subgraph_url": "https://api.studio.thegraph.com/query/57238/mech/version/latest"
+        "subgraph_url": ""
```
Comment (DavidMinarsch) [inline mech_client/configs/mechs.json:30]: keep missing values empty for clarity

File: mech_client/configs/mechs.json [hunk 9]
```diff
@@ -42,12 +42,12 @@
...
         "gas_limit": 100000,
         "contract_abi_url": "https://api.polygonscan.com/api?module=contract&action=getabi&address={contract_address}",
-        "subgraph_url": "https://api.studio.thegraph.com/query/57238/mech/version/latest"
+        "subgraph_url": ""
     },
     "base": {
         "agent_registry_contract": "0x88DE734655184a09B70700aE4F72364d1ad23728",
         "rpc_url": "https://base.llamarpc.com",
-        "wss_endpoint": "wss://base-rpc.publicnode.com",
+        "wss_endpoint": "wss://base.gateway.tenderly.co",
```
Comment (DavidMinarsch) [inline mech_client/configs/mechs.json:50]: other one doesn't work

File: mech_client/configs/mechs.json [hunk 10]
```diff
@@ -56,8 +56,8 @@
             "is_gas_estimation_enabled": false
         },
         "gas_limit": 100000,
-        "contract_abi_url": "https://base.blockscout.com/api/v2/smart-contracts/{contract_address}",
-        "subgraph_url": "https://api.studio.thegraph.com/query/57238/mech/version/latest"
+        "contract_abi_url": "https://api.basescan.org/api?module=contract&action=getabi&address={contract_address}",
```
Comment (DavidMinarsch) [inline mech_client/configs/mechs.json:59]: other one doesn't work

File: mech_client/subgraph.py [hunk 11]
```diff
@@ -62,9 +62,21 @@ def query_agent_address(
     :type url: str
     :param timeout: Timeout for the request.
     :type timeout: Optional[float]
+    :type chain_config: Optional[str]:
     :return: The agent address if found, None otherwise.
     :rtype: Optional[str]
     """
+    # temporary hard coded until subgraph present
```
Comment (DavidMinarsch) [inline mech_client/subgraph.py:69]: temporary

**Review by DavidMinarsch — APPROVED** [review]
merging this for now onto main

**Pull request merged**